### PR TITLE
implement button class

### PR DESCRIPTION
**Pull request by harry7557558**
Implement a button class.

The button class has an `__init__` function, a `draw` function, a `mouse_move` function and a `mouse_click` function to be called manually when the mouse moved or button-down. It supports different appearances when the mouse hovers or is clicked. It also allows adding a list of functions to be called when the mouse is clicked, which can be accessed using the `onclicks` object or the `add_onclick` instance.

### CODE REVIEW

File: rpg/button.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+class Button:
+    pos_x: float
+    pos_y: float
+    width: float
+    height: float
+    image_default: pygame.Surface
+    image_click: pygame.Surface
+    _image: pygame.Surface
+    onclicks: List[Callable]
```
Comment (MrGallo) [inline rpg/button.py]: Delete these.

File: rpg/button.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+    def __init__(self, x: float, y: float, w: float, h: float,
+                 image_default=None,
+                 image_hover=None,
+                 image_click=None):
```
Comment (MrGallo) [inline rpg/button.py]: Make `image_default` a required argument. Remove the `w` and `h` and derive them.

File: rpg/button.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        """Test if mouse position is on the button
+
+        Args:
+            mouse_x: x-coordinate of the mouse
+            mouse_y: y-coordinate of the mouse
+
+        Returns:
+            True if mouse is on the button, False otherwise
+        """
+        if 0 <= mouse_x-self.pos_x < self.width and 0 <= mouse_y-self.pos_y < self.height:
```
Comment (MrGallo) [inline rpg/button.py]: Suggestion: 
```python
return self.rect.collidepoint((mouse_x, mouse_y))
```

File: rpg/button.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+    def __init__(self, x: float, y: float, w: float, h: float,
+                 image_default=None,
+                 image_hover=None,
+                 image_click=None):
+        self.pos_x = x
+        self.pos_y = y
+        self.width = w
+        self.height = h
+        self.onclicks = []
```
Comment (MrGallo) [inline rpg/button.py:41]: Suggestion: Add `self.rect = pygame.Rect(x, y, w, h)`.

**Review by MrGallo — CHANGES_REQUESTED** [review]

**Comment by harry7557558** [comment]
I updated the constructor of the `Button` class. I made `image_default` a required argument, which can be a path to the image file, a `pygame.Surface`, or an RGB color. I keep `width` and `height` as optional arguments because they are needed when `image_default` is a color.

**Pull request closed without merging**